bayes_opt/bayesian_optimization.py

```diff
@@ -101,8 +110,26 @@ class BayesianOptimization(Observable):
     set_bounds()
         Allows changing the lower and upper searching bounds
     """
+
     def __init__(self, f, pbounds, random_state=None, verbose=2,
```

Comment:
Since `f` is useless when using a "full dataset" (i.e. reading both the X's and the target y's), I would add `None` default values to `f` and `pbounds`. We are forced to do this for `pbounds` due to how optional arguments work in Python. You can fix this by raising an error in the constructor if `pbounds=None`.

Speaking of errors, you should also add some other checks in this constructor. Namely, an error should be raised if both `f` and `target_column` are set, and if `target_column` is set but not `dataset_path`. Or anything else that comes to your mind. In short, we should manually handle (with sensible error messages) everything that would not work anyway.

Reply:
since the TargetSpace's attribute "params" depends on pbounds and it is used in the added functionalities, is it a good idea to set the default value to none?

Reply:
That's why I said:
> You can fix this by raising an error in the constructor if `pbounds=None`.

The code should always immediately stop if `pbounds=None` :)

Reply:
fixed